include/execution.hpp

```diff
@@ -1692,6 +1676,111 @@ namespace std::execution {
       }
     } transfer {};
   } // namespace __transfer
+
+  inline namespace __on {
+    namespace __impl {
+      template <class R_>
+        struct __receiver_ref : receiver_adaptor<__receiver_ref<R_>> {
+          using R = __t<R_>;
+          R& __r_;
+          R&& base() && noexcept { return (R&&) __r_; }
+          const R& base() const & noexcept { return __r_; }
+        };
+
+      template <class Sch_, class S_>
+        struct __sender : sender_base {
```

Comment:
In my current implementation I am specializing `sender_traits` for the inner sender
```cpp
    template <scheduler _Scheduler, sender _InnerSender>
    struct sender_traits<_On::_OuterSender<_Scheduler, _InnerSender>>
        : decltype(_Get_sender_traits_base<_InnerSender>()) {};
```

Would that be a valid alternative

Reply:
Your sender_traits specialization must dispatch to sender_traits on the inner receiver. There may also be a specialization of sender_traits for that sender type, and your suggested implementation would miss it.

Reply:
Thanks for pointing that out! Totally missed it